packages/block-library/src/social-links/edit.js

```diff
@@ -1,43 +1,124 @@
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+
 /**
  * WordPress dependencies
  */
 
 import { Fragment } from '@wordpress/element';
 
 import {
+	BlockControls,
 	__experimentalUseInnerBlocksProps as useInnerBlocksProps,
 	useBlockProps,
 	InspectorControls,
 } from '@wordpress/block-editor';
-import { ToggleControl, PanelBody } from '@wordpress/components';
+import {
+	Dropdown,
+	MenuGroup,
+	MenuItem,
+	PanelBody,
+	ToggleControl,
+	ToolbarButton,
+	ToolbarGroup,
+} from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
+import { check } from '@wordpress/icons';
+import { DOWN } from '@wordpress/keycodes';
 
 const ALLOWED_BLOCKS = [ 'core/social-link' ];
 
+const sizeOptions = [
+	{ name: __( 'Small' ), value: 'has-small-icon-size' },
+	{ name: __( 'Normal' ), value: 'has-normal-icon-size' },
+	{ name: __( 'Large' ), value: 'has-large-icon-size' },
+	{ name: __( 'Huge' ), value: 'has-huge-icon-size' },
+];
+
 export function SocialLinksEdit( props ) {
 	const {
-		attributes: { openInNewTab },
+		attributes: { iconSize, openInNewTab },
 		setAttributes,
 	} = props;
 
 	const SocialPlaceholder = (
 		<div className="wp-block-social-links__social-placeholder">
-			<div className="wp-block-social-link wp-social-link-facebook"></div>
-			<div className="wp-block-social-link wp-social-link-twitter"></div>
-			<div className="wp-block-social-link wp-social-link-instagram"></div>
+			<div className="wp-social-link wp-social-link-facebook"></div>
+			<div className="wp-social-link wp-social-link-twitter"></div>
+			<div className="wp-social-link wp-social-link-instagram"></div>
 		</div>
 	);
 
-	const blockProps = useBlockProps();
+	const className = classNames( iconSize );
+	const blockProps = useBlockProps( { className } );
 	const innerBlocksProps = useInnerBlocksProps( blockProps, {
 		allowedBlocks: ALLOWED_BLOCKS,
 		orientation: 'horizontal',
 		placeholder: SocialPlaceholder,
 		templateLock: false,
 		__experimentalAppenderTagName: 'li',
 	} );
+
+	const openOnArrowDown = ( event ) => {
```

Comment:
Mmm, why is this needed here? We should not have these kind of definitions in the blocks themselves but rely instead on the toolbar components handling it

Reply:
The event is fired when pressing the down arrow on the button, this is not the normal behavior for a button, but is natural (expected) in this case since it is opening a Dropdown menu, it will open and continuing down arrow will select the items.

I can see the need for a new type of button here that includes a dropdown, this behavior is also used in the Heading block for selecting level.

Testing again, and now it doesn't work as expected. The down arrow does open the menu, but pressing again does not move the selector, it loses focus.

Reply:
We have many controls in the toolbar that operate like dropdowns, seems it should be just handled with the core components. cc @diegohaz if you can give it a quick look

Reply:
@mkaz The generic `Dropdown` component doesn't add the key down handler automatically. `DropdownMenu` does. Have you tried it? For reference, the block switcher button uses it: https://github.com/WordPress/gutenberg/blob/99d02c51487c992699eab87c1c7ace90284d60e0/packages/block-editor/src/components/block-switcher/index.js#L159-L178

In case it helps, there's also this simple example on the developer handbook: https://developer.wordpress.org/block-editor/components/toolbar-item/#render-prop

Reply:
Thanks @diegohaz, I've got it working using `DropdownMenu` but some of the same issues persist with that component, also seen elsewhere, so I created a ticket to track. See #26401